src/main.rs

```diff
@@ -181,17 +178,42 @@ fn main() {
 
             assert!(success);
 
-            // Serialize proof into JSON
-            let json_proof = serde_json::to_string(&proof).unwrap();
+            match fs::create_dir("out") {
```

Comment:
You might want to use https://doc.rust-lang.org/stable/std/fs/fn.create_dir_all.html

for a local directory it makes no difference, but if you end up parametrizing this directory it might

Reply:
oh ok, will fix